src/profile-logic/comparison.js

```diff
@@ -142,9 +180,22 @@ export function mergeProfiles(
     resultProfile.threads.push(thread);
   }
 
+  if (resultProfile.threads.length === 2) {
```

Comment:
I always wanted to be able to import more than 2 profiles, and this is why the function is more generic than needed. Also that's because most of the function can be generic in a very easy way (we do exactly the same thing for both profiles, except when diffing).
Of course diffing can be done only for 2 profiles. But in the future, once we move this to selectors, we could possibly decide to add a "diffing track" out of the many tracks that have been added.

Reply:
I added a small comment explaining why we have this condition.